core/src/main/java/com/sequenceiq/cloudbreak/service/blueprint/DefaultBlueprintChecker.java

```diff
@@ -0,0 +1,53 @@
+package com.sequenceiq.cloudbreak.service.blueprint;
+
+import static com.sequenceiq.authorization.resource.AuthorizationResourceAction.DESCRIBE_CLUSTER_TEMPLATE;
+import static java.util.Collections.emptyList;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.authorization.resource.AuthorizationResourceAction;
+import com.sequenceiq.authorization.resource.AuthorizationResourceType;
+import com.sequenceiq.authorization.service.CrnsByCategory;
+import com.sequenceiq.authorization.service.DefaultResourceChecker;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.common.ResourceStatus;
+import com.sequenceiq.cloudbreak.domain.Blueprint;
+
+@Component
+public class DefaultBlueprintChecker implements DefaultResourceChecker {
+
+    @Inject
+    private BlueprintService blueprintService;
+
+    @Override
+    public AuthorizationResourceType getResourceType() {
+        return AuthorizationResourceType.CLUSTER_TEMPLATE;
+    }
+
+    @Override
+    public boolean isDefault(String resourceCrn) {
+        Blueprint b = blueprintService.getByResourceCrn(resourceCrn);
+        return b != null && ResourceStatus.DEFAULT == b.getStatus();
+    }
+
+    @Override
+    public boolean isAllowedAction(AuthorizationResourceAction action) {
+        return DESCRIBE_CLUSTER_TEMPLATE.equals(action);
+    }
+
+    @Override
+    public CrnsByCategory getDefaultResourceCrns(Collection<String> resourceCrns) {
+        Map<Boolean, List<String>> byDefault = resourceCrns.stream().collect(Collectors.partitioningBy(this::isDefault));
```

Comment:
Can you please do something like this:
```
SELECT just_the_necessary_fields WHERE crn in (:crns);
```
instead of

```
for ... {
  SELECT one where crn = :crn
}
```

Reply:
ok